src/lib/options/validate-options.ts

```diff
@@ -0,0 +1,170 @@
+  boolOpts.forEach((opt) => {
+    if (!is(options[opt], 'boolean')) return;
+    newOptions[opt] = options[opt];
+  });
+
+  // Validate number options
+  numberOpts.forEach((opt) => {
+    if (!is(options[opt], 'number')) return;
+    newOptions[opt] = options[opt];
+  });
+
+  // Validate string options
+  strOpts.forEach((opt) => {
+    if (!is(options[opt], 'string')) return;
+    newOptions[opt] = options[opt];
+  });
+
+  // Validate string or number options
+  strOrNumberOpts.forEach((opt) => {
+    if (!is(options[opt], 'string') && !is(options[opt], 'number')) return;
+    newOptions[opt] = options[opt];
+  });
+
+  // Validate function or string options
+  strOrFuncOpts.forEach((opt) => {
+    if (!is(options[opt], 'string') && !is(options[opt], 'function')) return;
+    newOptions[opt] = options[opt];
+  });
+
+  // Validate one of options
+  if (includes(validDataShapes, options.dataShape)) {
+    newOptions.dataShape = options.dataShape;
+  }
+
+  if (includes(validDataTypes, options.dataType)) {
+    newOptions.dataType = options.dataType;
+  }
+
+  if (includes(validLabelsPositions, options.labelsPosition)) {
+    newOptions.labelsPosition = options.labelsPosition;
+  }
+
+  if (includes(validControlButtons, options.controlButtons)) {
+    newOptions.controlButtons = options.controlButtons;
+  }
+
+  if (includes(validOverlays, options.overlays)) {
+    newOptions.overlays = options.overlays;
+  }
+
+  if (includes(validFillDateGapsIntervals, options.fillDateGapsInterval)) {
+    newOptions.fillDateGapsInterval = options.fillDateGapsInterval;
+  }
+
+  if (includes(validFillDateGapsValues, options.fillDateGapsValue)) {
+    newOptions.fillDateGapsValue = options.fillDateGapsValue;
+  }
+
+  // Validate array of options
+  if (is(options.fixedOrder, 'array', 'string')) {
+    newOptions.fixedOrder = options.fixedOrder;
+  }
+
+  // Other validations
+  if (validateDataTransform(options.dataTransform)) {
+    newOptions.dataTransform = options.dataTransform;
+  }
+
+  validateColorMap(options.colorMap);
+
+  return newOptions;
+}
+
+function validateDataTransform(value: Options['dataTransform'] | undefined): boolean {
+  if (typeof value === 'undefined') return false;
+
+  if (value === null || typeof value === 'function') {
+    return true;
+  }
+
+  return false;
+}
+
+function validateColorMap(value: string[] | { [key: string]: string } | undefined): boolean {
+  if (typeof value === 'undefined') return false;
+  if (value === null) return false;
+
+  // Check if color map is array of string
+  if (is(value, 'array', 'string')) return true;
+
+  if (is(value, 'object')) {
+    for (const [k, v] of Object.entries(value)) {
+      if (!is(k, 'string')) return false;
+      if (!is(v, 'string')) return false;
+    }
+  }
+
+  return false;
+}
+
+type types = 'array' | 'boolean' | 'object' | 'number' | 'string' | 'undefined' | 'function';
+export function is(value: any, type: types, arrayType?: types): boolean {
```

Comment:
This can also be simplified to:
```ts
function is(value: any, type: types, arrayType?: types): boolean {
  if (type === 'number') {
    return typeof value === 'number' && !isNaN(Number(value));
  }

  if (type === 'boolean') {
    return typeof value === 'boolean' || value === 'true' || value === 'false';
  }

  if (type === 'array') {
    return (
      Array.isArray(value) &&
      arrayType != null &&
      value.every((val) => is(val, arrayType))
    );
  }

  if (type === 'object') {
    return value != null && typeof value === 'object';
  }

  return typeof value === type;
}
```